js/components/keys/plate-color-key.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import ModelStore from "../../stores/model-store";
+import Plate1Color from "../../../images/plate-1-color.svg";
+import Plate2Color from "../../../images/plate-2-color.svg";
+import Plate2ColorAlt from "../../../images/plate-2-color-alt.svg";
+import Plate3Color from "../../../images/plate-3-color.svg";
+import Plate3ColorAlt from "../../../images/plate-3-color-alt.svg";
+import Plate4Color from "../../../images/plate-4-color.svg";
+import Plate5Color from "../../../images/plate-5-color.svg";
+
+import css from "../../../css-modules/keys/map-type.less";
+
+const plateCountToPlateColors: any[][] = [
+  [Plate1Color],
+  [Plate1Color, Plate2Color],
+  [Plate1Color, Plate2ColorAlt, Plate3ColorAlt],
+  [Plate1Color, Plate2Color, Plate3Color, Plate4Color],
+  [Plate1Color, Plate2Color, Plate3Color, Plate4Color, Plate5Color],
+];
+interface IProps {
+  model: ModelStore;
+}
+
+export const PlateColorKey = ({ model }: IProps) => {
```

Comment:
This component will not update itself / re-render when number of plates is changed. You can see it here:
https://tectonic-explorer.concord.org/branch/plate-color/index.html?rocks&preset=plates2
These two plates gets merged relatively soon (no forces, no motion). You'll see either 1 or 2 plates in the key, depending on when you open it. Also, resetting a model won't reset the key.

I think it's again the problem with functional components vs MobX. Probably adding `observer` will fix that, but I'm worried about passing various observables via props directly. This slightly defeats MobX's advantage of isolated re-render of the children without re-rendering of the parent. 

This is definitely not a task for this PR, but I think it'd be great to setup a complete pattern for MobX and functional components ASAP. We're supposed to have a ~week for maintenance, so I guess it'd be the perfect time for that.